kedro/runner/runner.py

```diff
@@ -443,3 +546,27 @@ def run_node(
     )
     node = task.execute()
     return node
+
+
+def validate_max_workers(max_workers: int | None) -> int:
```

Comment:
Is there a reason why `_validate_catalog`  and `_validate_nodes` are private methods and this one is public?

Reply:
I did that because `_validate_catalog` and `_validate_nodes` need to be implemented on the Runner level and aren't relevant to all runners atm, while `validate_max_workers` is a helper method used in both `ThreadRunner` and `ParallelRunner`. I can change it if it doesn't make sense.

Reply:
It feels odd to have this one exposed, although I appreciate that importing a private one in a couple of places also feels odd. Could we make it private in a non-disruptive way? Maybe a class method (or maybe not, since it's used in the ParallelRunnerManager)?

Reply:
Changed this to a private class method.